### PR TITLE
XML to JSON [11078]

### PR DESCRIPTION
**Pull request by tohar-gilbar**

### CODE REVIEW

File: lib/cveparser/parser.rb [hunk 1]
```diff
@@ -80,34 +72,34 @@ def self.matches_filter?(entry)
...
       params = {}
-      params[:cve] = entry.attributes['id'].value
+      params[:cve] = entry.dig("cve", "CVE_data_meta", "ID")
       params[:vulnerable_configurations] = vulnerable_configurations(entry)
       params[:cvss] = cvss(entry)
-      params[:vulnerable_software] = vulnerable_software(entry)
-      params[:published_datetime] = child_value(entry, 'vuln|published-datetime')
-      params[:last_modified_datetime] = child_value(entry, 'vuln|last-modified-datetime')
+      # in the xml format there was some duplication (configuration, software). now it appears once.
+      params[:vulnerable_software] = params[:vulnerable_configurations]
```
Comment (goldshani) [inline lib/cveparser/parser.rb]: You can delete the vulnerable_software(entry) method if it's not used anymore

File: lib/cveparser/parser.rb [hunk 2]
```diff
@@ -124,36 +116,42 @@ def self.vulnerable_software entry
...
-      entry.css('vuln|vulnerable-configuration > cpe-lang|logical-test'+
-                      ' > cpe-lang|fact-ref').each do |conf|
-          v_confs << conf.attributes['name'].value
+      entry.dig('configurations', 'nodes').each do |node|
+        if node.keys.include?('children')
+          node['children'].each do |child|
+            child['cpe_match'].each {|x| v_confs << x['cpe23Uri']}
+          end
+        else
+          node['cpe_match'].each {|x| v_confs << x['cpe23Uri']}
```
Comment (goldshani) [inline lib/cveparser/parser.rb]: Does each configuration node always have a 'cpe_match'?

File: lib/cveparser/rails_store.rb [hunk 3]
```diff
@@ -73,8 +73,8 @@ def self.save_entry(entry, with_products_hash)
           :cve => entry.cve,
           :cwe => entry.cwe,
           :summary => entry.summary,
-          :published => DateTime.xmlschema(entry.published_datetime),
-          :last_modified => DateTime.xmlschema(entry.last_modified_datetime),
+          :published => DateTime.parse(entry.published_datetime),
```
Comment (goldshani) [inline lib/cveparser/rails_store.rb]: Do the values you pass to DateTime.parse can be nil?

File: lib/cveparser/parser.rb (hunk 1 above)
Comment (tohar-gilbar) [reply lib/cveparser/parser.rb]: will do. deleted some other unused functions as well.

File: lib/cveparser/parser.rb (hunk 2 above)
Comment (tohar-gilbar) [reply lib/cveparser/parser.rb]: Sometimes they have `children` sometimes they have `cpe_match`.
The children have `cpe_match`

File: lib/cveparser/rails_store.rb (hunk 3 above)
Comment (tohar-gilbar) [reply lib/cveparser/rails_store.rb]: No

**Pull request merged**